subprojects/files/src/main/java/org/gradle/internal/file/impl/DefaultDeleter.java

```diff
@@ -152,29 +158,40 @@ private boolean shouldRemoveContentsOf(File file, Handling handling) {
         return file.isDirectory() && (handling.shouldFollowLinkedDirectory() || !isSymlink.test(file));
     }
 
-    protected boolean deleteFile(File file) {
-        return file.delete() && !file.exists();
+    protected boolean deleteFile(File file) throws IOException {
+        Path path = file.toPath();
+        return Files.deleteIfExists(path) && !Files.exists(path);
     }
 
     private boolean tryHardToDelete(File file) {
-        if (deleteFile(file)) {
-            return true;
-        }
-
-        // This is copied from Ant (see org.apache.tools.ant.util.FileUtils.tryHardToDelete).
-        // It mentions that there is a bug in the Windows JDK implementations that this is a valid
-        // workaround for. I've been unable to find a definitive reference to this bug.
-        // The thinking is that if this is good enough for Ant, it's good enough for us.
-        if (runGcOnFailedDelete) {
-            System.gc();
```

Comment:
❌ This is an unneeded change of behaviour

I would keep `runGcOnFailedDelete` as is and not limit retries to Windows.

Reply:
❌ Please restore the handling of `runGcOnFailedDelete`

Reply:
Thanks for the review.
Do I need to restore the comments here or what?

Reply:
Yes please